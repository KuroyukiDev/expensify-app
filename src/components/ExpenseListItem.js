import React from 'react';
import {Link} from 'react-router-dom';

const ExpenseListItem = ({id, desc, note, amount, createdAt}) => (
  
  <Link style={{textDecoration: 'none'}} to={`/edit/${id}`}>
    <div className="expense-list-item--text-content">
      <h4>Desc: {desc}</h4>
      <h5>Amount: {amount}</h5>
      <p>
        <strong>Note:</strong> <br/>
        {note}
      </p>
      <p> Created: {createdAt}</p>
      <br/>
      <hr/>
    </div>
  </Link>
);

export default ExpenseListItem;