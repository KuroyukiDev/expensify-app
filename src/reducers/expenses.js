
// Expenses Reducer
import {
  ADD_EXPENSE,
  EDIT_EXPENSE,
  REMOVE_EXPENSE
} from "../shared/types";

const expensesReducerDefaultState = [];

export default (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case ADD_EXPENSE:
      return [
        ...state,
        action.expense
      ];
    case REMOVE_EXPENSE:
      return state.filter(({ id }) => {
        return id !== action.id;
      });
    case EDIT_EXPENSE:
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    default:
      return state;
  }
};
