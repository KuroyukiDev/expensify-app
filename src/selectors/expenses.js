import moment from 'moment';
// Get visible expenses

export default (expenses, {
  text, sortBy, sortMode, startDate, endDate
}) => {
  return expenses.filter((expense) => {
    const createdAtMoment = moment(expense.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
    const textMatch = expense.desc.toLowerCase().includes(text.toLowerCase());
    
    // Returns a sorted version of the expenses array
    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if (sortMode === 'hi2lo') {
      if (sortBy === 'date') {
        return a.createdAt < b.createdAt ? 1 : -1;
      } else {
        return a.amount < b.amount ? 1 : -1;
      }
    } else {
      if (sortBy === 'date') {
        return a.createdAt > b.createdAt ? 1 : -1;
      } else {
        return a.amount > b.amount ? 1 : -1;
      }
    }
  });
};
