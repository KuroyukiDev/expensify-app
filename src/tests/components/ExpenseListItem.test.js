import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../../shared/testExpensesArr';

test('should render ExpenseListItem correctly', () => {
  const key = expenses[0].id;
  const wrapper = shallow(<ExpenseListItem key={key} {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});