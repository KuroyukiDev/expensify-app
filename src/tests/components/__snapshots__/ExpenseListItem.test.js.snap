// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<Link
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/edit/1"
>
  <div
    className="expense-list-item--text-content"
  >
    <h4>
      Desc: 
      Gum
    </h4>
    <h5>
      Amount: 
      $1.95
    </h5>
    <p>
      <strong>
        Note:
      </strong>
       
      <br />
    </p>
    <p>
      Created: 
      December 31st, 1969
    </p>
    <br />
    <hr />
  </div>
</Link>
`;
